ios/Classes/AVController.m

```diff
@@ -0,0 +1,63 @@
+#import "AVController.h"
+#import "Utility.h"
+#import <AVFoundation/AVFoundation.h>
+#import <MobileCoreServices/MobileCoreServices.h>
+
+@implementation AVController
+
++ (AVURLAsset *)getVideoAsset:(NSURL *)url {
+    return [AVURLAsset assetWithURL:url];
+}
+
++ (AVAssetTrack *)getTrack:(AVURLAsset *)asset {
+    __block AVAssetTrack* track = nil;
+    dispatch_group_t group = dispatch_group_create();
+    dispatch_group_enter(group);
+    NSArray *array = [NSArray arrayWithObject:@"tracks"];
+    [asset loadValuesAsynchronouslyForKeys:array completionHandler:^(void) {
+        AVKeyValueStatus status = [asset statusOfValueForKey:@"tracks" error:nil];
+        if (status == AVKeyValueStatusLoaded) {
+            track = [asset tracksWithMediaType:AVMediaTypeVideo].firstObject;
+        }
+        dispatch_group_leave(group);
+    }];
+    dispatch_group_wait(group, DISPATCH_TIME_FOREVER);
+    return track;
+}
+
++ (NSInteger)getVideoOrientation:(NSString *)path {
+    NSURL *url = [Utility getPathUrl:path];
+    AVURLAsset *asset = [self getVideoAsset:url];
+    AVAssetTrack *track = nil;
+    if ([self getTrack:asset]) {
+        track = [self getTrack:asset];
+    }
+    CGSize size = track.naturalSize;
+    CGAffineTransform txf = track.preferredTransform;
+    NSInteger orientation = 0;
+    if (size.width == txf.tx && size.height == txf.ty) {
+        orientation = 0;
+    } else if (txf.tx == 0 && txf.ty == 0) {
+        orientation = 90;
+    } else if (txf.tx == 0 && txf.ty == size.width) {
+        orientation = 180;
+    } else {
+        orientation = 270;
+    }
+    return orientation;
+}
+
++ (NSString *)getMetaDataByTag:(AVAsset *)asset key:(NSString *)key {
+    for (AVMetadataItem* item in asset.commonMetadata) {
+        if (item.commonKey.accessibilityValue == key) {
+            if (item.stringValue == (id)[NSNull null] || item.stringValue.length == 0 ) {
+                return item.stringValue;
+            } else {
+                return @"";
+            }
```

Comment:
I'm not sure I understand the purpose here. I checked the original code and the idea is to return `stringValue` or an empty string if it' not existent. Here you check if string is NSNull (for some reason) or length is 0 in order to return string value?
Shouldn't this be more like
```objc
if (item.stringValue != NULL) {
    return item.stringValue;
} else {
    return @"";
}
```
?

Can also be simplified to
```objc
return item.stringValue != NULL ? item.stringValue : @"";
```

Also, I don't think `item.commonKey.accessibilityValue` is what you are looking here for, since its not a raw value of the enum string representation in objective-c.
Did you test this code?

Reply:
I'm not sure about this part because I always get empty string for both objective-c and android,
what's your suggest for replacing item.commonKey.accessibilityValue (I cannot find rawValue)

Reply:
Changed to AVMetadataCommonKeyAuthor and AVMetadataCommonKeyTitle

Reply:
If we go by mimicking the original code you are taking from swift, it should be like this:
```objc
if ([item.commonKey isEqualToString:key]) {
    return item.stringValue != NULL ? item.stringValue : @"";
}
```

In objc commonKey is an NSString instance instead of an enum, so you can just compare it to another instance of the same class.